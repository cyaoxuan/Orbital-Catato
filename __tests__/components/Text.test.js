import {
    BodyText,
    ErrorText,
    RequiredFormText,
    TitleText,
} from "../../app/components/TextComponents";
import { cleanup, render } from "@testing-library/react-native";

afterEach(cleanup);

describe("<BodyText />", () => {
    it("renders successfully", () => {
        const { getByText } = render(<BodyText text="Hello" />);
        const bodyText = getByText("Hello");
        expect(bodyText).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<BodyText />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("<ErrorText />", () => {
    it("renders successfully", () => {
        const { getByText } = render(<ErrorText text="Error" />);
        const errorText = getByText("Error");
        expect(errorText).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<ErrorText />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("<RequiredFormText />", () => {
    it("renders successfully", () => {
        const { getByText } = render(<RequiredFormText text="Required" />);
        const requiredFormText = getByText("*Required");
        expect(requiredFormText).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<RequiredFormText />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("<TitleText />", () => {
    it("renders successfully", () => {
        const { getByText } = render(<TitleText text="Title" />);
        const titleText = getByText("Title");
        expect(titleText).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<TitleText />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
