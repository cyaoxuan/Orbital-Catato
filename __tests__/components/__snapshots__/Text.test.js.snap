// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BodyText /> renders correctly 1`] = `
<Text
  style={
    [
      {
        "textAlign": "left",
      },
      {
        "color": "rgba(28, 27, 31, 1)",
        "fontFamily": "System",
        "fontWeight": "400",
        "letterSpacing": 0,
      },
      {
        "writingDirection": "ltr",
      },
      {
        "color": undefined,
        "fontFamily": "Nunito-Medium",
      },
    ]
  }
/>
`;

exports[`<ErrorText /> renders correctly 1`] = `
<Text
  style={
    [
      {
        "textAlign": "left",
      },
      {
        "color": "rgba(28, 27, 31, 1)",
        "fontFamily": "System",
        "fontWeight": "400",
        "letterSpacing": 0,
      },
      {
        "writingDirection": "ltr",
      },
      {
        "color": "#BA1A1A",
        "fontFamily": "Nunito-Medium",
      },
    ]
  }
/>
`;

exports[`<RequiredFormText /> renders correctly 1`] = `
<Text
  style={
    [
      {
        "textAlign": "left",
      },
      {
        "color": "rgba(28, 27, 31, 1)",
        "fontFamily": "System",
        "fontWeight": "400",
        "letterSpacing": 0,
      },
      {
        "writingDirection": "ltr",
      },
      {
        "color": "#BA1A1A",
        "fontFamily": "Nunito-Bold",
      },
    ]
  }
>
  *
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "fontFamily": "Nunito-Medium",
        },
      ]
    }
  />
</Text>
`;

exports[`<TitleText /> renders correctly 1`] = `
<Text
  style={
    [
      {
        "textAlign": "left",
      },
      {
        "color": "rgba(28, 27, 31, 1)",
        "fontFamily": "System",
        "fontWeight": "400",
        "letterSpacing": 0,
      },
      {
        "writingDirection": "ltr",
      },
      {
        "color": undefined,
        "fontFamily": "Nunito-Bold",
      },
    ]
  }
/>
`;
